modules/rostests/apitests/shell32/ShellInfo.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Test for SHGetFileInfo
+ * COPYRIGHT:   Copyright 2024 Whindmar Saksit <whindsaks@proton.me>
+ */
+
+#include "shelltest.h"
+#include <shellutils.h>
+
+#define UNIQUEEXT L"ABC123XYZ"
+#define my_ok_all_flags(val, flags) ok_eq_hex((val) & (flags), (flags))
+
+static DWORD_PTR SHGFI(LPCWSTR Path, SHFILEINFOW &Info, UINT Flags, UINT Attributes = 0)
+{
+    return SHGetFileInfoW(Path, Attributes, &Info, sizeof(Info), Flags);
+}
+
+static DWORD_PTR SHGFI(LPCITEMIDLIST Pidl, SHFILEINFOW &Info, UINT Flags, UINT Attributes = 0)
+{
+    return SHGFI((PCWSTR)Pidl, Info, Flags | SHGFI_PIDL, Attributes);
+}
+
+START_TEST(SHGetFileInfo)
+{
+    CCoInit ComInit;
+    LPITEMIDLIST pidl;
+    SHFILEINFOW info;
+    UINT flags;
+    WCHAR buf[MAX_PATH];
+
+    ok_int(SHGFI((PCWSTR)NULL, info, 0), FALSE);
```

Comment:
Is the cast done on NULL actually necessary?

Reply:
Yes, SHGFI is overloaded to also take a PIDL.